gpytorch/kernels/kernel.py

```diff
@@ -318,18 +341,13 @@ def _covar_dist(
                     res = res.pow(2)
             if postprocess:
                 res = dist_postprocess_func(res)
+            return res
 
         elif square_dist:
             res = self.distance_module._jit_sq_dist(x1, x2, torch.tensor(x1_eq_x2), postprocess)
         else:
             res = self.distance_module._jit_dist(x1, x2, torch.tensor(x1_eq_x2), postprocess, torch.tensor(False))
 
-        if batch_dims == (0, 2):
-            if diag:
-                res = res.transpose(0, 1).contiguous().view(-1, res.shape[-1])
-            else:
-                res = res.transpose(0, 1).contiguous().view(-1, *res.shape[-2:])
-
```

Comment:
Same here. I think this line is still necessary for additive structure/SKIP.

Reply:
Not with the new multibatch way of handling batch_dims